go-controller/pkg/util/multi_network_test.go

```diff
@@ -845,6 +845,121 @@ func TestGetPodNADToNetworkMapping(t *testing.T) {
 	}
 }
 
+func TestSubnetOverlapCheck(t *testing.T) {
+	_, cidr4, _ := net.ParseCIDR("10.128.0.0/14")
+	_, cidr6, _ := net.ParseCIDR("fe00::/16")
+	_, svcCidr4, _ := net.ParseCIDR("172.30.0.0/16")
+	_, svcCidr6, _ := net.ParseCIDR("fe01::/16")
+	config.Default.ClusterSubnets = []config.CIDRNetworkEntry{{cidr4, 24}, {cidr6, 64}}
+	config.Kubernetes.ServiceCIDRs = []*net.IPNet{svcCidr4, svcCidr6}
+	config.Gateway.V4MasqueradeSubnet = "169.254.169.0/29"
+	config.Gateway.V6MasqueradeSubnet = "fd69::/125"
+	config.Gateway.V4JoinSubnet = "100.64.0.0/16"
+	config.Gateway.V6JoinSubnet = "fd98::/64"
+	type testConfig struct {
+		desc                        string
+		inputNetAttachDefConfigSpec string
+		expectedError               error
+	}
+
```

Comment:
can have one +ve test with no expectedError

Reply:
I have kepthttps://github.com/ovn-org/ovn-kubernetes/pull/4604/commits/41e41c69c59b89d41d2cfa9d1f5d1147ec68c0f6#diff-9c47311a9d19959866ca074985bfd6523244f714186bb068c910ac22013fdc28R927 , should i add more? I can do that if required.

Reply:
ok, that is enough.